server/data_cxg/cxg_adaptor.py

```diff
@@ -51,8 +51,13 @@ def set_tiledb_context(context_params):
         """Set the tiledb context.  This should be set before any instances of CxgAdaptor are created"""
         try:
             CxgAdaptor.tiledb_ctx = tiledb.Ctx(context_params)
+            tiledb.default_ctx(context_params)
+
         except tiledb.libtiledb.TileDBError as e:
-            raise ConfigurationError(f"Invalid tiledb context: {str(e)}")
+            if e.message == "Global context already initialized!":
```

Comment:
This is likely only triggered in unit tests where we would initialize more than once, which would be okay.  But it would be a bug if this happened outside of testing,  and in this implementation it would be a silent bug.  

The only real problem is if we try to change the tiledb config after we've already set the default_ctx.  

I suggest doing this:

```
if e.message == "Global context already initialized!":
   if tiledb.default_ctx().config().dict() != CxgAdaptor.tiledb_ctx.config().dict():
      raise ConfigurationError(f"Cannot change tiledb configuration once it is set")
else:
    raise ConfigurationError(f"Invalid tiledb context: {str(e)}")
```

Now the only problem would be if we had a test file where we wanted to set the tiledb config differently in two different tests.  I don't think we have that, but if we did we could split those into different test files, or find some other workaround.

Reply:
👍